app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt

```diff
@@ -40,12 +43,12 @@ class NewConversationViewModel
     private val searchKnownUsers: SearchKnownUsersUseCase,
     private val searchPublicUsers: SearchUserDirectoryUseCase,
     private val getAllContacts: GetAllContactsUseCase,
-    private val createGroupConversation: CreateGroupConversationUseCase
+    private val createGroupConversation: CreateGroupConversationUseCase,
+    private val getSelf: GetSelfUserUseCase,
 ) : ViewModel() {
 
-    // TODO: map this value out with the given back-end configuration later on
     private companion object {
-        const val HARDCODED_TEST_DOMAIN = "wire.com"
+        val FEDERATION_REGEX = Regex("[^@.]+@[^@.]+\\.[^@]+")
```

Comment:
I'm not 100% sure if this regex is 100% correct because I don't know whole logic for generating them, but it expects 
1. any length of characters w/o "@" and "." 
2. then expects "@" 
3. then expects any length of characters w/o "@" and "." 
4. then expects "."
5. finally any length of characters

this is a super simple regex, because even if no i.e special characters are not allowed and user will search for it will always returns no results and I think it's indifferent if federated or not federated search will return no results because result will be expected anyway

Reply:
perhaps covering it with unit tests will help us answer this question ?